Source/WebGPU/WebGPU/CommandEncoder.mm

```diff
@@ -734,6 +779,14 @@ static bool validateCopyTextureToBuffer(const WGPUImageCopyTexture& source, cons
         // "When you copy to a 2D texture, depth must be 1."
         auto sourceSize = MTLSizeMake(widthForMetal, heightForMetal, 1);
         auto sourceOrigin = MTLOriginMake(source.origin.x, source.origin.y, 0);
+        if (!widthForMetal || !heightForMetal)
+            return;
+
+        if (widthForMetal * heightForMetal > destinationBuffer.length) {
```

Comment:
I think this is wrong, at least it causes a regression in compressed texture formats copyTextureToTexture cases